homeassistant/components/tplink/quality_scale.yaml

```diff
@@ -0,0 +1,66 @@
+rules:
+  # Bronze
+  config-flow: done
+  test-before-configure: done
+  unique-config-entry: done
+  config-flow-test-coverage: done
+  runtime-data: done
+  test-before-setup: done
+  appropriate-polling: done
+  entity-unique-id: done
+  has-entity-name: done
+  entity-event-setup:
+    status: exempt
+    comment: The integration does not use events.
+  dependency-transparency: todo
+  action-setup:
+    status: exempt
+    comment: The integration only uses platform services.
+  common-modules: done
+  docs-high-level-description: done
+  docs-installation-instructions: done
+  docs-removal-instructions: done
+  docs-actions: done
+  brands: done
+
+  # Silver
+  config-entry-unloading: done
+  log-when-unavailable: done
+  entity-unavailable: done
+  action-exceptions: todo
+  reauthentication-flow: done
+  parallel-updates: todo
+  test-coverage: done
+  integration-owner: done
+  docs-installation-parameters: todo
+  docs-configuration-parameters:
+    status: exempt
+    comment: The integration does not have any options configuration parameters.
+
+  # Gold
+  entity-translations: done
+  entity-device-class: done
+  devices: done
+  entity-category: done
+  entity-disabled-by-default: done
+  discovery: done
+  stale-devices: todo
+  diagnostics: done
+  exception-translations: done
+  icon-translations: done
+  reconfiguration-flow: done
+  dynamic-devices: todo
```

Comment:
I believe some have this but some do not yet right? Should we note down what still needs to happen here?

Reply:
I think this is related to hub-connected child devices. The upstream library doesn't currently handle cases where a new device (like thermostat) is added or if one gets de-paired.